frontend/src/app/components/ExperimentPage.js

```diff
@@ -48,14 +48,15 @@ export default class ExperimentPage extends React.Component {
 
   componentWillReceiveProps(nextProps) {
     const { shouldShowTourDialog, enabled: prevEnabled } = this.state;
-    const { isExperimentEnabled, getExperimentBySlug } = nextProps;
 
-    const prevExperiment = getExperimentBySlug(this.props.params.slug);
+    const prevExperiment = this.props.getExperimentBySlug(this.props.params.slug);
     const prevInProgress = prevExperiment && prevExperiment.inProgress;
 
-    const nextExperiment = getExperimentBySlug(nextProps.params.slug);
+    const { isExperimentEnabled: nextIsExperimentEnabled,
+            getExperimentBySlug: nextGetExperimentBySlug } = nextProps;
```

Comment:
actually, there's something unsettling to me about having these functions as props at all. I haven't dug into it but it *feels* like this component should have an `experiment` prop instead. anyway... maybe a change for another day